dio/lib/src/transformer.dart

```diff
@@ -57,25 +56,32 @@ abstract class Transformer {
   }
 }
 
+/// The callback definition for decoding a JSON string.
+typedef JsonDecodeCallback = FutureOr<dynamic> Function(String);
+
+/// The callback definition for encoding a JSON object.
+typedef JsonEncodeCallback = FutureOr<String> Function(Object);
+
 /// The default [Transformer] for [Dio].
 ///
 /// If you want to custom the transformation of request/response data,
 /// you can provide a [Transformer] by your self, and replace
 /// the [DefaultTransformer] by setting the [dio.transformer].
-
-typedef JsonDecodeCallback = dynamic Function(String);
-
 class DefaultTransformer extends Transformer {
-  DefaultTransformer({this.jsonDecodeCallback});
+  DefaultTransformer({
```

Comment:
Regarding changing the default transformer to the new background one, can we deprecate this by type alias, maybe `SynchronizedTransformer` or `ForegroundTransformer` (v.s. `BackgroundTransformer`)?
```dart
@Deprecated('Use BackgroundTransformer instead')
typedef DefaultTransformer = SynchronizedTransformer;
```

Reply:
Sure, will do 👍

Reply:
Done. I've also moved the old `DefaultTransformer` into the transformer folder.